### DISCUSSION TITLE
Option to disable LWT reporting for DeepSleep devices (Home Assistant integration)

### DISCUSSION THREAD

**Discussion by macrojames** [arendst/Tasmota#16031] [Ideas]
### PROBLEM DESCRIPTION
`DEEPSLEEP_LWT_HA_DISCOVERY` does not work with new Home Assistant integration (from https://github.com/arendst/Tasmota/pull/8724)
It was a fix for #8655, a problem that is back when using the new method over legacy discovery

I can't prevent the battery operated devices to show "unavailable" on Home Assistant while the devices are deep sleeping.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Generic
- [x] Tasmota binary firmware version number used: 12.0.2
  - [x] Pre-compiled
  - [x] Self-compiled
- [x] Flashing tools used: PIO
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
Use any sensor device, send command `DeepsleepTime 90` and see it **go unavailable** in Home Assistant.

### EXPECTED BEHAVIOUR
Use any sensor device, send command `DeepsleepTime 90` and see it **retain** its value in Home Assistant, at least for "DeepSleepTime length seconds"

### SCREENSHOTS
<img width="244" alt="image" src="https://user-images.githubusercontent.com/482151/179849152-92756360-4dc4-4528-b4b5-9d0e4326a3d7.png">
(I was debugging -> different intervals)

### ADDITIONAL CONTEXT
I think this is still a design issue with the usage of LWT messages and deep sleeping devices.
Any remote thermometer or sth like that will show the last received value for some time, maybe hours after the sender went offline.

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by macrojames** [comment]
Working earlier, but not with current integration is a Feature Request?

**Comment by ascillato** [comment]
It is a Feature Request because it was never implemented in the new integration. There were several discussions about this and the solution back then was to config everything manual in HA when having deep sleep. Your request is to make Deep Sleep be supported for the new HA integration (or at least LWT configurable). The previous auto discovery is deprecated and not everything has been added to the new one. Anyone interested on programming it and providing a pull request is more than welcome.

**Comment by stefanbode** [comment]
We are working on this right now. Also battery level will be supported soon.

**Reply by stefanbode** [reply]
"batterylevel" is now supported but still the LWT does not work. The Tasmota team did not like the idea of introducing a third state "online/offline/Deepsleep"

Still work in progress

**Reply by stefanbode** [reply]
there will be a new update 0.8.0 hatasmota now support deepsleep. Device will not go unavailable anymore on deepsleep. Thanks to Erik @emontnemery
